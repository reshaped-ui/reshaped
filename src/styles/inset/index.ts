import { responsiveVariables } from "utilities/helpers";
import * as T from "styles/types";
import * as G from "types/global";
import "./inset.css";

const getInsetStyles: (
	value?: G.Responsive<number | "auto">,
	side?: "top" | "bottom" | "start" | "end"
) => T.VariableStyleUtilityResult = (value, side) => {
	if (value === undefined) return null;
	const suffix = side ? `-${side}` : "";
	const variableName = `--rs-inset${suffix}` as const;
	const variables = responsiveVariables(variableName, value);

	return { variables };
};

export default getInsetStyles;
