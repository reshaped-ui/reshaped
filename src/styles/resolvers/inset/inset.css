@responsive [style*="--rs-inset-s:"] {
	@variable --rs-inset 0;

	inset: calc(var(--rs-inset) * var(--rs-unit-x1));
}

@responsive [style*="--rs-inset-end-"] {
	@variable --rs-inset-end 0;

	inset-inline-end: calc(var(--rs-inset-end) * var(--rs-unit-x1));
}

@responsive [style*="--rs-inset-start-"] {
	@variable --rs-inset-start 0;

	inset-inline-start: calc(var(--rs-inset-start) * var(--rs-unit-x1));
}

@responsive [style*="--rs-inset-top-"] {
	@variable --rs-inset-top 0;

	inset-block-start: calc(var(--rs-inset-top) * var(--rs-unit-x1));
}

@responsive [style*="--rs-inset-bottom-"] {
	@variable --rs-inset-bottom 0;

	inset-block-end: calc(var(--rs-inset-bottom) * var(--rs-unit-x1));
}
