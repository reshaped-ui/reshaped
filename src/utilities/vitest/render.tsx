import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
