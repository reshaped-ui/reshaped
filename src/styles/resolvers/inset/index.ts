import { responsiveVariables } from "utilities/props";
import * as T from "styles/types";
import "./inset.css";

const inset: T.StyleResolver<T.Inset> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-inset", value) };
};

export const insetTop: T.StyleResolver<T.Inset> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-inset-top", value) };
};

export const insetBottom: T.StyleResolver<T.Inset> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-inset-bottom", value) };
};

export const insetStart: T.StyleResolver<T.Inset> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-inset-start", value) };
};

export const insetEnd: T.StyleResolver<T.Inset> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-inset-end", value) };
};

export const insetInline: T.StyleResolver<T.Inset> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-inset-inline", value) };
};

export const insetBlock: T.StyleResolver<T.Inset> = (value) => {
	if (value === undefined) return {};
	return { variables: responsiveVariables("--rs-inset-block", value) };
};

export default inset;
