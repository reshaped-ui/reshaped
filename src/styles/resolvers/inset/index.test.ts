import { expect, test, describe } from "vitest";
import inset, {
	insetTop,
	insetBottom,
	insetStart,
	insetEnd,
	insetInline,
	insetBlock,
} from "./index";

describe("Styles/Inset", () => {
	test("handles positive value", () => {
		expect(inset(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(inset(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(inset()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(inset({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Inset/Top", () => {
	test("handles positive value", () => {
		expect(insetTop(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(insetTop(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(insetTop()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(insetTop({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Inset/Bottom", () => {
	test("handles positive value", () => {
		expect(insetBottom(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(insetBottom(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(insetBottom()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(insetBottom({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Inset/Start", () => {
	test("handles positive value", () => {
		expect(insetStart(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(insetStart(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(insetStart()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(insetStart({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Inset/End", () => {
	test("handles positive value", () => {
		expect(insetEnd(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(insetEnd(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(insetEnd()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(insetEnd({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Inset/Inline", () => {
	test("handles positive value", () => {
		expect(insetInline(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(insetInline(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(insetInline()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(insetInline({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});

describe("Styles/Inset/Block", () => {
	test("handles positive value", () => {
		expect(insetBlock(4)).toMatchSnapshot();
	});

	test("handles 0 value", () => {
		expect(insetBlock(0)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(insetBlock()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(insetBlock({ s: 4, m: 0, l: 2 })).toMatchSnapshot();
	});
});
