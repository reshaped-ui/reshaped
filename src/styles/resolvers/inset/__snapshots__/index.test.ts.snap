// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/Inset > handles 0 value 1`] = `
{
  "variables": {
    "--rs-inset-s": 0,
  },
}
`;

exports[`Styles/Inset > handles positive value 1`] = `
{
  "variables": {
    "--rs-inset-s": 4,
  },
}
`;

exports[`Styles/Inset > handles responsive value 1`] = `
{
  "variables": {
    "--rs-inset-l": 2,
    "--rs-inset-m": 0,
    "--rs-inset-s": 4,
  },
}
`;

exports[`Styles/Inset > handles undefined value 1`] = `{}`;

exports[`Styles/Inset/Block > handles 0 value 1`] = `
{
  "variables": {
    "--rs-inset-block-s": 0,
  },
}
`;

exports[`Styles/Inset/Block > handles positive value 1`] = `
{
  "variables": {
    "--rs-inset-block-s": 4,
  },
}
`;

exports[`Styles/Inset/Block > handles responsive value 1`] = `
{
  "variables": {
    "--rs-inset-block-l": 2,
    "--rs-inset-block-m": 0,
    "--rs-inset-block-s": 4,
  },
}
`;

exports[`Styles/Inset/Block > handles undefined value 1`] = `{}`;

exports[`Styles/Inset/Bottom > handles 0 value 1`] = `
{
  "variables": {
    "--rs-inset-bottom-s": 0,
  },
}
`;

exports[`Styles/Inset/Bottom > handles positive value 1`] = `
{
  "variables": {
    "--rs-inset-bottom-s": 4,
  },
}
`;

exports[`Styles/Inset/Bottom > handles responsive value 1`] = `
{
  "variables": {
    "--rs-inset-bottom-l": 2,
    "--rs-inset-bottom-m": 0,
    "--rs-inset-bottom-s": 4,
  },
}
`;

exports[`Styles/Inset/Bottom > handles undefined value 1`] = `{}`;

exports[`Styles/Inset/End > handles 0 value 1`] = `
{
  "variables": {
    "--rs-inset-end-s": 0,
  },
}
`;

exports[`Styles/Inset/End > handles positive value 1`] = `
{
  "variables": {
    "--rs-inset-end-s": 4,
  },
}
`;

exports[`Styles/Inset/End > handles responsive value 1`] = `
{
  "variables": {
    "--rs-inset-end-l": 2,
    "--rs-inset-end-m": 0,
    "--rs-inset-end-s": 4,
  },
}
`;

exports[`Styles/Inset/End > handles undefined value 1`] = `{}`;

exports[`Styles/Inset/Inline > handles 0 value 1`] = `
{
  "variables": {
    "--rs-inset-inline-s": 0,
  },
}
`;

exports[`Styles/Inset/Inline > handles positive value 1`] = `
{
  "variables": {
    "--rs-inset-inline-s": 4,
  },
}
`;

exports[`Styles/Inset/Inline > handles responsive value 1`] = `
{
  "variables": {
    "--rs-inset-inline-l": 2,
    "--rs-inset-inline-m": 0,
    "--rs-inset-inline-s": 4,
  },
}
`;

exports[`Styles/Inset/Inline > handles undefined value 1`] = `{}`;

exports[`Styles/Inset/Start > handles 0 value 1`] = `
{
  "variables": {
    "--rs-inset-start-s": 0,
  },
}
`;

exports[`Styles/Inset/Start > handles positive value 1`] = `
{
  "variables": {
    "--rs-inset-start-s": 4,
  },
}
`;

exports[`Styles/Inset/Start > handles responsive value 1`] = `
{
  "variables": {
    "--rs-inset-start-l": 2,
    "--rs-inset-start-m": 0,
    "--rs-inset-start-s": 4,
  },
}
`;

exports[`Styles/Inset/Start > handles undefined value 1`] = `{}`;

exports[`Styles/Inset/Top > handles 0 value 1`] = `
{
  "variables": {
    "--rs-inset-top-s": 0,
  },
}
`;

exports[`Styles/Inset/Top > handles positive value 1`] = `
{
  "variables": {
    "--rs-inset-top-s": 4,
  },
}
`;

exports[`Styles/Inset/Top > handles responsive value 1`] = `
{
  "variables": {
    "--rs-inset-top-l": 2,
    "--rs-inset-top-m": 0,
    "--rs-inset-top-s": 4,
  },
}
`;

exports[`Styles/Inset/Top > handles undefined value 1`] = `{}`;
